Implemented docs code block component (#2973)
cfengine/documentation: cheatsheet.markdown

To turn on syntax highlighting, specify the language ("brush") directly after the opening three backticks.
Syntax highlighting is provided by pygments. Find all available lexers [here](http://pygments.org/docs/lexers/).

#### Command code blocks

```command
python3 -v
```

This code block will have `command` in the header and corresponding icon.

#### Command code block with output

To have a component that shows command, and it's output you need to place output code block following command one.

```command
uname
```

```output
Linux
```

You might also specify output syntax highlighting by adding language
after the starting backticks and placing `[output]` in the first line.
This line won't be shown in the resulted HTML.

```command
curl --user admin:admin https://test.cfengine.com/api/user
```

```json
[output]
{
  "meta": {
    "page": 1,
    "count": 1,
    "total": 1,
    "timestamp": 1350994249d
  },
  "data": [
    {
      "id": "calvin",
      "external": true,
      "roles": [
        "Huguenots", "Marketing"
      ]
    }
  ]
}
```

These two blocks will be joined into one element on the UI.

#### File code block

> You can specify file type of the code block by adding  `[file=Name of the file]` in the first line.

Reviewer: ```suggestion
You can specify file name of the code block by adding  `[file=Name of the file]` in the first line.
```

> This line won't be shown in the resulted HTML.

Reviewer 2: ```suggestion
This line won't be shown in the resulting HTML (it will be converted to the heading / frame).
```

```cf3

> [file=Agent.cf]

Reviewer 2: ```suggestion
[file=policy.cf]
```
